{"name": "Успешное добавление товара в корзину", "status": "failed", "statusDetails": {"message": "AssertionError: Товар не добавлен в корзину\nassert '1' == '2'\n  \n  - 2\n  + 1", "trace": "self = <web.tests.test_cart.TestCartPage object at 0x0000018CA501B2D0>, chrome_driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"a6c320a5870612ea50b1f1451cc677ed\")>\n\n    @allure.title(\"Успешное добавление товара в корзину\")\n    def test_add_to_cart(self, chrome_driver):\n        login_page = LoginPage(chrome_driver)\n        product_page = ProductPage(chrome_driver)\n        login_page.go_to_site()\n        login_page.login(Config.USERNAME, Config.PASSWORD)\n        product_page.add_product_to_cart()\n        try:\n>           assert (\n                product_page.find_element(\n                    (\"xpath\", \"//span[@class='shopping_cart_badge']\")\n                ).text\n                == \"2\"\n            ), \"Товар не добавлен в корзину\"\nE           AssertionError: Товар не добавлен в корзину\nE           assert '1' == '2'\nE             \nE             - 2\nE             + 1\n\nweb\\tests\\test_cart.py:21: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <web.tests.test_cart.TestCartPage object at 0x0000018CA501B2D0>, chrome_driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"a6c320a5870612ea50b1f1451cc677ed\")>\n\n    @allure.title(\"Успешное добавление товара в корзину\")\n    def test_add_to_cart(self, chrome_driver):\n        login_page = LoginPage(chrome_driver)\n        product_page = ProductPage(chrome_driver)\n        login_page.go_to_site()\n        login_page.login(Config.USERNAME, Config.PASSWORD)\n        product_page.add_product_to_cart()\n        try:\n            assert (\n                product_page.find_element(\n                    (\"xpath\", \"//span[@class='shopping_cart_badge']\")\n                ).text\n                == \"2\"\n            ), \"Товар не добавлен в корзину\"\n            chrome_driver.execute_script(\"window.localStorage.clear();\")\n        except AssertionError as e:\n>           self.handle_assertion_error(chrome_driver, e)\n\nweb\\tests\\test_cart.py:29: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <web.tests.test_cart.TestCartPage object at 0x0000018CA501B2D0>, chrome_driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"a6c320a5870612ea50b1f1451cc677ed\")>\nerror_message = AssertionError(\"Товар не добавлен в корзину\\nassert '1' == '2'\\n  \\n  - 2\\n  + 1\")\n\n    def handle_assertion_error(self, chrome_driver, error_message):\n        with allure.step(\"Поймали ошибку\"):\n            allure.attach(\n                chrome_driver.get_screenshot_as_png(),\n                name=\"screenshot.png\",\n                attachment_type=allure.attachment_type.PNG,\n            )\n>           raise AssertionError(error_message)\nE           AssertionError: Товар не добавлен в корзину\nE           assert '1' == '2'\nE             \nE             - 2\nE             + 1\n\nweb\\tests\\manager_test.py:14: AssertionError"}, "steps": [{"name": "Ввод имени пользователя: standard_user", "status": "passed", "steps": [{"name": "Click on ('xpath', \"//input[@id='user-name']\")", "status": "passed", "start": 1711123303969, "stop": 1711123303978}], "start": 1711123303969, "stop": 1711123304027}, {"name": "Ввод пароля: secret_sauce", "status": "passed", "steps": [{"name": "Click on ('xpath', \"//input[@id='password']\")", "status": "passed", "start": 1711123304027, "stop": 1711123304035}], "start": 1711123304027, "stop": 1711123304070}, {"name": "Нажатие кнопки входа", "status": "passed", "steps": [{"name": "Click on ('xpath', \"//input[@id='login-button']\")", "status": "passed", "start": 1711123304070, "stop": 1711123304079}], "start": 1711123304070, "stop": 1711123304136}, {"name": "Добавляем товар в корзину", "status": "passed", "steps": [{"name": "Click on ('xpath', \"(//button[text()='Add to cart'])[1]\")", "status": "passed", "start": 1711123304136, "stop": 1711123304145}], "start": 1711123304136, "stop": 1711123304171}, {"name": "Click on ('xpath', \"//span[@class='shopping_cart_badge']\")", "status": "passed", "start": 1711123304171, "stop": 1711123304179}, {"name": "Поймали ошибку", "status": "failed", "statusDetails": {"message": "AssertionError: Товар не добавлен в корзину\nassert '1' == '2'\n  \n  - 2\n  + 1\n", "trace": "  File \"C:\\Dev\\python_selenium\\web\\tests\\manager_test.py\", line 14, in handle_assertion_error\n    raise AssertionError(error_message)\n"}, "attachments": [{"name": "screenshot.png", "source": "229fc526-2ea6-4812-b948-61e055b43358-attachment.png", "type": "image/png"}], "start": 1711123304188, "stop": 1711123304358}], "start": 1711123303277, "stop": 1711123304359, "uuid": "75177843-4f7a-4e1f-85e3-a4c844e495db", "historyId": "372d59ffd5d2f5632326fe25ddd3e2b9", "testCaseId": "372d59ffd5d2f5632326fe25ddd3e2b9", "fullName": "web.tests.test_cart.TestCartPage#test_add_to_cart", "labels": [{"name": "feature", "value": "Работа с корзиной"}, {"name": "suite", "value": "Работа с корзиной"}, {"name": "tag", "value": "@pytest.mark.usefixtures('chrome_driver')"}, {"name": "parentSuite", "value": "web.tests"}, {"name": "subSuite", "value": "TestCartPage"}, {"name": "host", "value": "DESKTOP-AMCLGNE"}, {"name": "thread", "value": "14640-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "web.tests.test_cart"}]}